File: Backends/C#/SuperHeroApi/SuperHeroCore/ServiceResponse.cs
using System;
using System.Collections.Generic;
using System.Net;
using System.Text;

namespace SuperHeroCore
{
    public class ServiceResponse
    {
        public DateTime IssuedOn { get; set; } = DateTime.Now;
        public HttpStatusCode Status { get; set; }
        public string StatusDescription { get; set; }
        public void SetOk() => Status = HttpStatusCode.OK;
        public void SetInternalServerError() => Status = HttpStatusCode.InternalServerError;
    }
}

File: Backends/TypeScript/src/SuperHeroCore/ServiceResponse.ts
import HttpStatusCode from "./Config/HttpStatusCode";

export class ServiceResponse{

    issued_on: Date = new Date();
    status: HttpStatusCode = HttpStatusCode.I_AM_A_TEAPOT;
    status_description: string = "";
    
    public SetOk(): void { this.status = HttpStatusCode.OK };
    public SetInternalServerError(): void {this.status = HttpStatusCode.INTERNAL_SERVER_ERROR};
}

